### PR TITLE
[Fix] Moving apt from workflow to script

### PR DESCRIPTION
**Pull request by Yarboa**

### CODE REVIEW

File: tests/cpp_check.sh [hunk 1]
```diff
@@ -2,10 +2,19 @@
...
+  sudo apt install -y make
+  sudo apt install -y autoconf
+  sudo apt install -y automake
+  sudo apt install -y cppcheck
+fi
+ 
+if command -v cppcheck > /dev/null; then
+  u="$u -U __REDIRECT_NTH -U _BSD_RUNE_T_ -U _TYPE_size_t -U __LDBL_REDIR1_DECL"
+  supp="--suppress=missingInclude --suppress=unusedFunction"
+  suppf="redblack.c"
```
Comment (ericcurtin) [inline tests/cpp_check.sh:16]: This line can be removed, this file only exists in inotify-tools and the `| grep -v "$suppf"` part.

Comment (Yarboa) [reply tests/cpp_check.sh:16]: @ericcurtin 
If you look at this test result, i would like to reduce cppchecks warnings
It is accepted ? 
Would you prefer separate PR?

Comment (ericcurtin) [reply tests/cpp_check.sh:16]: I am happy to merge like this or wait until green. We certainly have to go through them and fix what's worthwhile, this warning can be turned off "noExplicitConstructor", that's one that's not worth fixing.

Comment (ericcurtin) [reply tests/cpp_check.sh:16]: Once this build is green we will stop merging things that are not green

**Comment by Yarboa** [comment]
Thanks. 
Will upload PR with other on other code errors

File: tests/cpp_check.sh [hunk 2]
```diff
@@ -2,10 +2,19 @@
...
+  sudo apt install -y make
+  sudo apt install -y autoconf
+  sudo apt install -y automake
+  sudo apt install -y cppcheck
+fi
+ 
+if command -v cppcheck > /dev/null; then
+  u="$u -U __REDIRECT_NTH -U _BSD_RUNE_T_ -U _TYPE_size_t -U __LDBL_REDIR1_DECL"
+  supp="--suppress=missingInclude --suppress=unusedFunction --suppress=noExplicitConstructor"
+  supp="$supp --suppress=useInitializationList --suppress=uninitMemberVar"
```
Comment (ericcurtin) [inline tests/cpp_check.sh:16]: I'll probably end up fixing these, useInitializationList and uninitMemberVar, I can remove this when I get around to it.

**Pull request merged**